streams.go

```diff
@@ -238,6 +233,12 @@ type baseStreamReader struct {
 
 	closed atomic.Bool
 	lock   sync.Mutex
+	cond   *sync.Cond
+
+	// readerMutex is used to lock the reader when reading the stream
+	// this will prevent issues when the user tries to read from multiple methods at the same time
+	// since cond unlocks the lock before waiting, this is needed to prevent multiple read calls from fighting for data
```

Comment:
can remove this comment, this feels like comment on how a condition variable works and not specific to this use case, would be good to remove non-specific comments.